src/components/tabs/tab-bar.tsx

```diff
@@ -1,131 +1,175 @@
+        previousElement.className = 'deanimated';
+        array_move(tabList, tabList.indexOf(dragged), tabList.indexOf(dragged) - 1);
+        setTabList([...tabList]);
       }
     }
-    elemn.onmouseleave = (e: any) => closeDragElement(e, elemn);
-
   }
 
-  function closeDragElement(e: any, elemn: any) {
-    e = e || window.event;
-    if (document.getElementById('new')) {
-      tabBar.current.removeChild(document.getElementById('new'));
-    }
-    elemn.style.position = 'relative';
-    elemn.style.left = 'auto';
-    elemn.style.width = '145px';
-    tabBar.current.onmouseup = null;
-    tabBar.current.onmousemove = null;
-    tabBar.current.onmouseleave = null;
-    const nodeList = tabBar.current.children;
-    for (let i = 0; i < nodeList.length; i += 1) {
-      if (nodeList[i].id !== e.target.id) {
-        nodeList[i].style.pointerEvents = 'all';
+  function closeDragElement(e: React.MouseEvent<HTMLElement>) {
+    if (dragged) {
+      const elemn = getRef(dragged).current;
+      const nextElement = elemn.nextSibling as HTMLElement;
+      const previousElement = elemn.previousSibling as HTMLElement;
+      if (nextElement) {
+        nextElement.style.marginLeft = `${0}px`;
+        nextElement.style.marginRight = `${0}px`;
+      }
+      if (previousElement) {
+        previousElement.style.marginLeft = `${0}px`;
+        previousElement.style.marginRight = `${0}px`;
       }
+      const element = tabList.find(item => item.props.id === 'new');
+      const index = tabList.indexOf(element);
+      if (element) {
+        tabList.splice(index, 1);
+      }
+      setTabList([...tabList]);
+      elemn.style.position = 'relative';
+      elemn.style.left = 'auto';
+      elemn.style.width = '145px';
+      tabBar.current.onmouseup = null;
+      draggingBack.current = false;
+      draggingFront.current = false;
+      setDrag(null);
     }
   }
 
-  // closes elements based on sibling
-  const removeTab = (id: string, e: any) => {
+  // closes elements based on List Order
+  const removeTab = (id: string, e: any, tab: any) => {
     e.stopPropagation();
-    const elemn = document.getElementById(id);
-    if (checkActive(id)) {
-      if (elemn.nextElementSibling) {
-        setActive((elemn.nextSibling as HTMLElement).id, e);
-      } else if (elemn.previousSibling) {
-        setActive((elemn.previousSibling as HTMLElement).id, e);
+    if (checkActive(tab.props.id)) {
+      if (tabList.length > 1) {
+        const backTab = tabList[tabList.indexOf(tab) + 1];
+        const frontTab = tabList[tabList.indexOf(tab) - 1];
+        if (backTab) {
+          setActive(backTab.props.id, e);
+        } else {
+          setActive(frontTab.props.id, e);
+        }
       }
-
     }
-
-    const tabNode = document.getElementById(id);
-    tabBar.current.removeChild(tabNode);
-
-    const panelNode = document.getElementById(`${id}-panel`);
-    panelNode.parentNode.removeChild(panelNode);
+    // finish this up
```

Comment:
Why?

Reply:
will remove this, makes no sense right now